docs/proposal/2025-01-07-aws-self-managed-feature-gates.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Feature gates for AWSCluster and AWSMachine
+authors:
+  - "@mzazrivec"
+reviewers:
+  - "@serngawy"
+  - "@nrb"
+creation-date: 2025-01-07
+last-updated: 2025-01-13
+status: provisional
+---
+
+# Feature gates for AWS self managed clusters
+
+## Table of Contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Glossary](#glossary)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals/Future Work](#non-goalsfuture-work)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Functional Requirements](#functional-requirements)
+- [Alternatives](#alternatives)
+- [Upgrade Strategy](#upgrade-strategy)
+- [Implementation History](#implementation-history)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+## Summary
+
+Feature gates are a mechanism that CAPA uses to enable or disable particular features components of CAPA. This document proposes to implement two new feature gates, one for AWSMachine controller, another one for AWSCluster controller.
+
+## Motivation
+
+Motivation for the two new feature gates is the option to turn on and off CAPA's ability to reconcile AWSMachine and AWSCluster resources for AWS self-managed clusters. Currently, controllers for these two resource types are always on.
```

Comment:
Do you mean EKS clusters? The words "self-managed", "managed" and "unmanaged" are quite ambiguous already and I'd avoid them.

Reply:
Fair enough, fixed.